### ISSUE TITLE
HubstaffClient locale::facet::_S_create_c_locale name not valid

### ISSUE THREAD

**Issue by srghma** [NixOS/nixpkgs#38988]
## Issue description
```sh
$ HubstaffClient
2018-04-16 10:55:13.565472 [DEBUG] [0x1d45490] ui_style_adwaita.cpp:437  DEfinineg adwaita
2018-04-16 10:55:13.565966 [DEBUG] [0x1d45490] fltk.cpp:218  Define scheme: adwaita
2018-04-16 10:55:13.566193 [DEBUG] [0x1d45490] fltk.cpp:230  Define scheme: adwaita first box: 64
2018-04-16 10:55:13.610665 [DEBUG] [0x1d45490] fltk.cpp:218  Define scheme: cairo
2018-04-16 10:55:13.610823 [DEBUG] [0x1d45490] fltk.cpp:230  Define scheme: cairo first box: 67
2018-04-16 10:55:13.610924 [DEBUG] [0x1d45490] fltk_cairo.cpp:565  cairo_box_start : 67
2018-04-16 10:55:13.611274 [DEBUG] [0x1d45490] Log.cpp:230  Echoing to stdout : 1
2018-04-16 10:55:13.611365 [DEBUG] [0x1d45490] main.cpp:20   Hubstaff client started
terminate called after throwing an instance of 'std::runtime_error'
  what():  locale::facet::_S_create_c_locale name not valid
[1]    10331            HubstaffClient
```

### Steps to reproduce
version - 1.3.1-ff75f26 (im using latest version from local nixpkgs clone, prev version of hubstaff has this issue too)

Though everything worked fine before latest update

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.33, NixOS, 18.03.131954.2569e482904 (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0`
 - channels(bjorn): `"nixos-17.09-17.09.3202.9a8344a7a72"`
 - channels(root): `"nixos-18.03.131954.2569e482904"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by srghma** [comment]
@michalrus

**Comment by srghma** [comment]
`sudo nixos-rebuild switch --rollback` have worked, hubstaff is working again in version 

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.33, NixOS, 18.03.131807.489a14add9a (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0`
 - channels(bjorn): `"nixos-17.09-17.09.3202.9a8344a7a72"`
 - channels(root): `"nixos-18.03.131954.2569e482904"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

```sh
HubstaffClient
2018-04-16 11:12:20.732812 [DEBUG] [0x28f6160] ui_style_adwaita.cpp:437  DEfinineg adwaita
2018-04-16 11:12:20.734214 [DEBUG] [0x28f6160] fltk.cpp:218  Define scheme: adwaita
2018-04-16 11:12:20.734355 [DEBUG] [0x28f6160] fltk.cpp:230  Define scheme: adwaita first box: 64
2018-04-16 11:12:20.784518 [DEBUG] [0x28f6160] fltk.cpp:218  Define scheme: cairo
2018-04-16 11:12:20.784590 [DEBUG] [0x28f6160] fltk.cpp:230  Define scheme: cairo first box: 67
2018-04-16 11:12:20.784621 [DEBUG] [0x28f6160] fltk_cairo.cpp:565  cairo_box_start : 67
2018-04-16 11:12:20.788767 [DEBUG] [0x28f6160] Log.cpp:230  Echoing to stdout : 1
2018-04-16 11:12:20.788806 [DEBUG] [0x28f6160] main.cpp:20   Hubstaff client started
2018-04-16 11:12:20.790590 [DEBUG] [0x28f6160] LinuxXGrabber.cpp:81   Enhanced Window Manager is present, name = i3
2018-04-16 11:12:20.791416 [DEBUG] [0x28f6160] LinuxXGrabber.cpp:320  id: termite name: Termite binary: /nix/store/n24q3rllnkpw9xaxgkl9bj77jps3wkqk-termite-13/bin/.termite-wrapped
2018-04-16 11:12:20.791796 [INFO] [0x28f6160] Log.cpp:321  ------------------------------------------
2018-04-16 11:12:20.792087 [INFO] [0x28f6160] Log.cpp:322  Log opened.
2018-04-16 11:12:20.792150 [INFO] [0x28f6160] Log.cpp:327  Local Time : 2018-04-16 11:12:20
2018-04-16 11:12:20.792193 [INFO] [0x28f6160] Log.cpp:328  UTC Time   : 2018-04-16 08:12:20
2018-04-16 11:12:20.792244 [INFO] [0x28f6160] Log.cpp:329  TZ Offset  : 03:00:00
2018-04-16 11:12:20.792274 [INFO] [0x28f6160] Log.cpp:330  ------------------------------------------
2018-04-16 11:12:20.792303 [INFO] [0x28f6160] Platform.cpp:103  Last shutdown was clean
2018-04-16 11:12:20.792334 [INFO] [0x28f6160] Platform.cpp:105  Client Version:  1.3.0-9b2ba62
2018-04-16 11:12:20.792364 [INFO] [0x28f6160] Platform.cpp:106  Client Platform: Linux x86_64
2018-04-16 11:12:20.794428 [INFO] [0x28f6160] Platform.cpp:107  Client OS:       Linux machine 4.14.33 #1-NixOS SMP Sun Apr 8 12:26:34 UTC 2018 x86_64 GNU/Linux
2018-04-16 11:12:20.794496 [INFO] [0x28f6160] Platform.cpp:108  Client MEM:
2018-04-16 11:12:20.794532 [INFO] [0x28f6160] Platform.cpp:109  Client Data:     /home/bjorn/.local/share/HubStaff
2018-04-16 11:12:20.794561 [INFO] [0x28f6160] Platform.cpp:110  ------------------------------------------
2018-04-16 11:12:20.796543 [DEBUG] [0x28f6160] Configuration.cpp:105  Loaded Config
```

I'm not sure what exactly caused this

**Comment by michalrus** [comment]
After looking through the diff, I don’t see what can cause this locale regression. Maybe let’s ping… @Mic92 @orivej @adisbladis @grahamc @fpletz

**Comment by Mic92** [comment]
Probably related to https://github.com/NixOS/nixpkgs/issues/38991

**Comment by michalrus** [comment]
@srghma what’s the status rn?

**Comment by michalrus** [comment]
Sad. :crying_cat_face: 

FWIW, this patch is enough to update, but on b32839211ba7727ed87cb2b8e4ec80f3b1006b84 (Mar 14, 2018):

```diff
diff --git a/pkgs/applications/misc/hubstaff/default.nix b/pkgs/applications/misc/hubstaff/default.nix
index 2b6e58a6ab3..364979b2c26 100644
--- a/pkgs/applications/misc/hubstaff/default.nix
+++ b/pkgs/applications/misc/hubstaff/default.nix
@@ -5,7 +5,7 @@
 
 let
 
-  version = "1.3.1-ff75f26";
+  version = "1.4.0-f3657dd";
 
   rpath = stdenv.lib.makeLibraryPath
     [ libX11 zlib libSM libICE libXext freetype libXrender fontconfig libXft
@@ -19,7 +19,7 @@ stdenv.mkDerivation {
 
   src = fetchurl {
     url = "https://hubstaff-production.s3.amazonaws.com/downloads/HubstaffClient/Builds/Release/${version}/Hubstaff-${version}.sh";
-    sha256 = "0jm5l34r6lkfkg8vsdfqbr0axngxznhagwcl9y184lnyji91fmdl";
+    sha256 = "1mcbkh2lvjm0rfybk9jj6izrp2xnm78zgq9f14y75kxw0v5jk0ml";
   };
 
   nativeBuildInputs = [ unzip makeWrapper ];
```

**Comment by srghma** [comment]
I have switched to 18.09 and suddenly patch on b328392 no longer working, dont understand why

**Comment by srghma** [comment]
working hubstaff package https://github.com/srghma/dotfiles/blob/5bc601f30c8a2666d3b2d3c7e995507e164eab31/nixos/pkgs/hubstaff/default.nix#L1

**Comment by michalrus** [comment]
Seems to me the issue is fixed with:
* nixpkgs 929cc78363e6878e044556bd291382eab37bcbed
* Hubstaff 1.4.3-b4b3cb24

**Comment by srghma** [comment]
@michalrus is link to commit https://github.com/NixOS/nixpkgs/commit/929cc78363e6878e044556bd291382eab37bcbed wrong?

**Comment by fpletz** [comment]
I can confirm that this is fixed on master and thus for 19.03.
